src/main/java/hudson/plugins/copyartifact/Copier.java

```diff
@@ -35,7 +35,7 @@
      * @param baseTargetDir Base target dir for upcoming file copy (the copy-artifact
      *   build step may later specify a deeper target dir)
      */
-    public abstract void init(Run src, AbstractBuild<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
+    public abstract void init(Run src, Run<?,?> dst, FilePath srcDir, FilePath baseTargetDir) throws IOException, InterruptedException;
```

Comment:
Possible backward-compatibility issue. I am not sure if there is really an expectation of an implementation of this extension point in another plugin, but if so, you need to do some gymnastics with `Util.isOverridden` to make it compatible to both call and override the original overload.